amr-wind/equation_systems/icns/source_terms/GeostrophicForcing.cpp

```diff
@@ -30,26 +31,35 @@ GeostrophicForcing::GeostrophicForcing(const CFDSim& /*unused*/)
         amrex::ParmParse pp("CoriolisForcing");
         amrex::Real rot_time_period = 86400.0;
         pp.query("rotational_time_period", rot_time_period);
+        // Latitude is mandatory, everything else is optional
+        // Latitude is read in degrees
+        pp.get("latitude", latitude);
         coriolis_factor = 2.0 * utils::two_pi() / rot_time_period;
         amrex::Print() << "Geostrophic forcing: Coriolis factor = "
                        << coriolis_factor << std::endl;
-
-        amrex::Real latitude = 90.0;
-        pp.query("latitude", latitude);
-        AMREX_ALWAYS_ASSERT(
-            amrex::Math::abs(latitude - 90.0) <
-            static_cast<amrex::Real>(vs::DTraits<float>::eps()));
     }
 
     {
         // Read the geostrophic wind speed vector (in m/s)
         amrex::ParmParse pp("GeostrophicForcing");
         pp.getarr("geostrophic_wind", m_target_vel);
```

Comment:
add this `pp.query("three_ComponentForcing", m_three_dimensional_forcing);` under GeostrophicForcing

Reply:
I'm not sure this is what I want. The way I currently have it:
amrex::ParmParse pp("ABL");
        pp.query("three_ComponentForcing", m_three_dimensional_forcing);

Makes the user set the number of forcing components under the ABL section, which I think makes sense.